File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        currsum = 0
        maxi = nums[0]

        for val in nums:
            currsum += val
            maxi = max(currsum, maxi)
            if(currsum < 0) : currsum = 0
        return maxi
        

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& arr) {
        int n = arr.size();

        int sum = 0;
        int maxi = INT_MIN;

        for(int val : arr) {
            sum += val;
            maxi = max(maxi, sum);

            if(sum < 0) sum = 0;
        }

        return maxi;
        
    }
};